#
# Copyright (c) 2012, 2015, Oracle and/or its affiliates. All rights reserved.
# This patch includes the fix for:
# Bug 16807596 - [XKB] Sun Type 7 Keyboard Scroll lock and
#    Compose indicators (LEDs) don't work
--- a/rules/base.xml.in	Tue Feb 26 15:32:19 2013
+++ b/rules/base.xml.in	Wed Feb 17 15:31:29 2016
@@ -1458,6 +1458,12 @@
             <_description>English (Workman, international with dead keys)</_description>
            </configItem>
          </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>English (US, Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -1592,6 +1598,12 @@
         </variant>
         <variant>
           <configItem>
+            <name>sun_type6</name>
+            <_description>Arabic (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
+        <variant>
+          <configItem>
             <name>mac</name>
             <_description>Arabic (Macintosh)</_description>
           </configItem>
@@ -1790,6 +1802,12 @@
             <_description>Belgian (Wang model 724 azerty)</_description>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>Belgian (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -2272,6 +2290,12 @@
             </languageList>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>Portuguese (Brazil, Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -2567,6 +2591,12 @@
             </languageList>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>Multilingual (Canada, Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -2702,6 +2732,12 @@
         </variant>
         <variant>
           <configItem>
+            <name>sun_type6</name>
+            <_description>Czech (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
+        <variant>
+          <configItem>
             <name>rus</name>
             <!-- Keyboard indicator for Russian layouts -->
             <_shortDescription>ru</_shortDescription>
@@ -2754,6 +2790,12 @@
             <_description>Danish (Dvorak)</_description>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>Danish (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -2785,6 +2827,12 @@
             <_description>Dutch (standard)</_description>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>Dutch (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -2827,6 +2875,12 @@
             <_description>Estonian (US keyboard with Estonian letters)</_description>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>Estonian (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -3015,6 +3069,12 @@
         </variant>
         <variant>
           <configItem>
+            <name>sun_type6</name>
+            <_description>Finnish (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
+        <variant>
+          <configItem>
             <name>das</name>
             <_description>Finnish (DAS)</_description>
           </configItem>
@@ -3134,6 +3194,12 @@
             </languageList>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>French (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -3413,6 +3479,12 @@
         </variant>
         <variant>
           <configItem>
+            <name>sun_type6</name>
+            <_description>German (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
+        <variant>
+          <configItem>
             <name>legacy</name>
             <_description>German (legacy)</_description>
           </configItem>
@@ -3454,6 +3526,12 @@
             <_description>Greek (polytonic)</_description>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>Greek (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -3707,6 +3785,12 @@
             <_description>Italian (IBM 142)</_description>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>Italian (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -3750,6 +3834,24 @@
             <_description>Japanese (Dvorak)</_description>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>Japanese (Sun Type 6)</_description>
+          </configItem>
+        </variant>
+        <variant>
+          <configItem>
+            <name>sun_type7</name>
+            <_description>Japanese (Sun Type 7 - pc compatible)</_description>
+          </configItem>
+        </variant>
+        <variant>
+          <configItem>
+            <name>sun_type7_suncompat</name>
+            <_description>Japanese (Sun Type 7 - sun compatible)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -3941,6 +4043,12 @@
             <_description>Lithuanian (LEKPa)</_description>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>Lithuanian (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -3990,6 +4098,12 @@
             <_description>Latvian (adapted)</_description>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>Latvian (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -4176,6 +4290,12 @@
             <_description>Norwegian (Colemak)</_description>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>Norwegian (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -4254,6 +4374,12 @@
             <_description>Polish (programmer Dvorak)</_description>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>Polish (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -4318,6 +4444,12 @@
             </languageList>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>Portuguese (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -4355,6 +4487,12 @@
             <_description>Romanian (WinKeys)</_description>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>Romanian (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -4521,6 +4659,12 @@
         </variant>
         <variant>
           <configItem>
+            <name>sun_type6</name>
+            <_description>Russian (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
+        <variant>
+          <configItem>
             <name>phonetic_azerty</name>
             <_description>Russian (phonetic azerty)</_description>
           </configItem>
@@ -4651,6 +4795,12 @@
             <_description>Slovak (qwerty, extended Backslash)</_description>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>Slovak (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -4718,6 +4868,12 @@
             <_description>Spanish (Macintosh)</_description>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>Spanish (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -4795,6 +4951,12 @@
             </languageList>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>Swedish (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -4883,6 +5045,18 @@
             <_description>German (Switzerland, Macintosh)</_description>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6_de</name>
+            <_description>German (Switzerland, Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6_fr</name>
+            <_description>French (Switzerland, Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -5132,6 +5306,12 @@
             </languageList>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>Turkish (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -5235,6 +5415,12 @@
             <_description>Ukrainian (homophonic)</_description>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>Ukrainian (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -5290,6 +5476,12 @@
             <_description>English (UK, Colemak)</_description>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>English (UK, Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -5340,6 +5532,12 @@
             <_description>Korean (101/104 key compatible)</_description>
           </configItem>
         </variant>
+        <variant>
+          <configItem>
+            <name>sun_type6</name>
+            <_description>Korean (Sun Type 6/7)</_description>
+          </configItem>
+        </variant>
       </variantList>
     </layout>
     <layout>
@@ -6297,6 +6495,19 @@
           <_description>Scroll Lock</_description>
         </configItem>
       </option>
+    </group>
+    <group allowMultipleSelection="true">
+      <!-- Using LEDs to indicate modifiers -->
+      <configItem>
+        <name>mod_led</name>
+        <_description>Use keyboard LED to indicate modifiers</_description>
+      </configItem>
+      <option>
+        <configItem>
+          <name>mod_led:compose</name>
+          <_description>Compose</_description>
+        </configItem>
+      </option>
     </group>
     <group allowMultipleSelection="false">
       <!-- Select a keypad type -->
